org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/JavacUtils.java

```diff
@@ -182,7 +182,10 @@ private static void configureOptions(IJavaProject javaProject, Context context,
 
 		Map<String, String> processorOptions = ProcessorConfig.getProcessorOptions(javaProject);
 		for (Entry<String, String> processorOption : processorOptions.entrySet()) {
-			options.put("-A" + processorOption.getKey() + "=" + processorOption.getValue(), Boolean.toString(true));
+			options.put("-A", processorOption.getKey() + "=" + processorOption.getValue());
```

Comment:
Can we also use `Options.A` here by the way?

Reply:
Options.A only enables annotation processing. We also need to add -Akey=value to all annotation processors.

Reply:
Option.A is just a richer definition for `-A`. Looking at how Option.A is consumed, I think it should be good to use it here too with `options.put(Option.A, processorOption.getKey() + '=' + processorOptions.getValue())`.

Reply:
Have you tested it with `options.put(Option.A, processorOption.getKey() + '=' + processorOptions.getValue())`? From my last experiment, `options.put("-A" + processorOption.getKey() + "=" + processorOption.getValue(), Boolean.toString(true));` works well.

Reply:
You can try the following:
- add `org.eclipse.jdt.apt.processorOptions/a=b` to ~/.settings/org.eclipse.jdt.apt.core.prefs
- apply the following patch
```
diff --git a/org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/JavacUtils.java b/org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/JavacUtils.java
index d52462a5b8..536879c434 100644
--- a/org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/JavacUtils.java
+++ b/org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/JavacUtils.java
@@ -184,6 +184,9 @@ public class JavacUtils {
                Map<String, String> processorOptions = ProcessorConfig.getProcessorOptions(javaProject);
                for (Entry<String, String> processorOption : processorOptions.entrySet()) {
                        options.put(Option.A, processorOption.getKey() + "=" + processorOption.getValue());
+                       if (!options.isSet(Option.A)) {
+                               ILog.get().error("Option.A is not set");
+                       }
                }
                if (!options.isSet(Option.A) && ProcessorConfig.isAnnotationProcessingEnabled(javaProject)) {
                        options.put(Option.A, "");
```
- recompile the project (Project>Clean)
- check the error log. You should not see `Option.A is not set`

`options.put(Option.A, processorOption.getKey() + '=' + processorOptions.getValue())` sets
`-A=key=value` - the javac compiler recognizes it
options.put("-A" + processorOption.getKey() + "=" + processorOption.getValue(), Boolean.toString(true)) sets
`-A=key=value=true` - the javac compiler doesn't recognize it

Reply:
You can use `options.put(Option.A,,..` or`options.put("-A,"...`
See https://github.com/openjdk/jdk/blob/40ae4699622cca72830acd146b7b5c4efd5a43ec/src/jdk.compiler/share/classes/com/sun/tools/javac/util/Options.java#L149C4-L155C6

Reply:
The javac compiler uses `Option.IsSet` at
https://github.com/openjdk/jdk/blob/e023addf701ce4321040c96bd501355ece75a05c/src/jdk.compiler/share/classes/com/sun/tools/javac/main/JavaCompiler.java#L1301-L1311

Reply:
> The javac compiler uses `Option.IsSet` at https://github.com/openjdk/jdk/blob/e023addf701ce4321040c96bd501355ece75a05c/src/jdk.compiler/share/classes/com/sun/tools/javac/main/JavaCompiler.java#L1301-L1311
> 
> ```
>     static boolean explicitAnnotationProcessingRequested(Options options, JavaFileManager fileManager) {
>         return
>             options.isSet(PROCESSOR) ||
>             options.isSet(PROCESSOR_PATH) ||
>             options.isSet(PROCESSOR_MODULE_PATH) ||
>             options.isSet(PROC, "only") ||
>             options.isSet(PROC, "full") ||
>             options.isSet(A) ||
>             options.isSet(XPRINT) ||
>             fileManager.hasLocation(ANNOTATION_PROCESSOR_PATH);
>         // Skipping -XprintRounds and -XprintProcessorInfo
> ```

This is not the code that will consume the processor options. The link below is how Javac compiler parses the processor options from the Options object.

https://github.com/openjdk/jdk/blob/e023addf701ce4321040c96bd501355ece75a05c/src/jdk.compiler/share/classes/com/sun/tools/javac/processing/JavacProcessingEnvironment.java#L639

The previous implementation in this PR https://github.com/eclipse-jdtls/eclipse-jdt-core-incubator/pull/820 should work well.

Reply:
You are right. 
JavacProcessingEnvironment.initProcessorOptions() expects -`Akey=value=true`. 
Options.isSet(Option.A) expects `-A=`
I have updated the PR.